### PR TITLE
15 pre backup hook

### PR DESCRIPTION
**Pull request by sbancal**
needed to backup servers with DB that need to be dumped before backup

### CODE REVIEW

File: src/enacrestic/restic_backup.py [hunk 1]
```diff
@@ -168,21 +176,32 @@ def _handle_stderr(self):
...
-            self.need_to_unlock = lock_hours_bool or lock_minutes_int > 30
-            if not self.need_to_unlock:
-                self.app.logger.write(
-                    "The lock is too new. We expect 30+ minutes to unlock manually."
+        if self.state.current_operation in (
+            Operation.BACKUP,
+            Operation.FORGET,
+            Operation.UNLOCK,
+        ):
+            if re.search(r"timeout", stderr):
```
Comment (guilbep) [inline src/enacrestic/restic_backup.py:184]: @sbancal On est obligé de faire des regex là ? y'a pas de code d'erreurs ?

File: src/enacrestic/state.py [hunk 2]
```diff
@@ -134,11 +138,18 @@ def get_icon(self):
...
             if self.current_status == Status.OK:
-                return (
-                    f"{const.ICONS_FOLDER}/backup_success_badge.png"
-                    if self.version_need_upgrade()
-                    else f"{const.ICONS_FOLDER}/backup_success.png"
-                )
+                if not self.pre_backup_failed:
+                    return (
+                        f"{const.ICONS_FOLDER}/backup_success_badge.png"
+                        if self.version_need_upgrade()
```
Comment (guilbep) [inline src/enacrestic/state.py:144]: @sbancal I find it difficult to read when we have more than 2 levels of if.

**Review by guilbep — APPROVED** [review]

**Comment by guilbep** [comment]
@blueur Do you have suggestions about the 4 levels of nested ifs ?

**Comment by sbancal** [comment]
@blueur @guilbep as said ... open for a discussion on coding style when it'll be a good time for this.
Right now, the feature is here and there are no remarks on the behavior, so I merge.
thks

**Pull request merged**